patches/minecraft/net/minecraft/entity/EntityLivingBase.java.patch

```diff
@@ -39,7 +39,24 @@
      }
  
      public EntityLivingBase func_110144_aD()
-@@ -812,6 +814,8 @@
+@@ -668,8 +670,14 @@
+         else
+         {
+             Collection<PotionEffect> collection = this.field_70713_bf.values();
+-            this.field_70180_af.func_187227_b(field_184634_g, Boolean.valueOf(func_184593_a(collection)));
+-            this.field_70180_af.func_187227_b(field_184633_f, Integer.valueOf(PotionUtils.func_185181_a(collection)));
++            boolean hideParticles = Boolean.valueOf(func_184593_a(collection));
++            int potionColor = Integer.valueOf(PotionUtils.func_185181_a(collection));
++            net.minecraftforge.event.entity.living.PotionColorCalculationEvent event = new net.minecraftforge.event.entity.living.PotionColorCalculationEvent(this, potionColor, hideParticles, collection);
++            net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(event);
++            hideParticles = event.isHideParticle();
++            potionColor = event.getColor();
++            this.field_70180_af.func_187227_b(field_184634_g, hideParticles);
++            this.field_70180_af.func_187227_b(field_184633_f, potionColor);
```

Comment:
Patches have to be as small as possible, even if it is a bit hard to read.
I think you can make this patch much smaller by avoiding variables like this:
```java
 ++            net.minecraftforge.event.entity.living.PotionColorCalculationEvent event = new net.minecraftforge.event.entity.living.PotionColorCalculationEvent(this, potionColor, func_184593_a(collection), PotionUtils.func_185181_a(collection));
 ++            net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(event);
 ++            this.field_70180_af.func_187227_b(field_184634_g, event.isHideParticle());
 ++            this.field_70180_af.func_187227_b(field_184633_f, event.getColor());
```